diagnostic_updater/diagnostic_updater/_diagnostic_updater.py

```diff
@@ -229,40 +229,22 @@ class Updater(DiagnosticTaskVector):
     reason.
     """
 
-    def __init__(self, node):
+    def __init__(self, node, period=1.0):
         """Construct an updater class."""
         DiagnosticTaskVector.__init__(self)
         self.node = node
         self.publisher = self.node.create_publisher(DiagnosticArray, '/diagnostics', 1)
         self.clock = Clock()
-        now = self.clock.now()
-
-        self.last_time = now
-
-        self.last_time_period_checked = self.last_time
         self.period_parameter = 'diagnostic_updater.period'
-        self.period = self.node.declare_parameter(self.period_parameter, 1.0).value
+        self.__period = self.node.declare_parameter(self.period_parameter, period).value
+        self.timer = self.node.create_timer(self.__period, self.update)
 
         self.verbose = False
         self.hwid = ''
         self.warn_nohwid_done = False
 
     def update(self):
         """Causes the diagnostics to update if the inter-update interval has been exceeded."""
-        self._check_diagnostic_period()
-        now = self.clock.now()
-        if now >= self.last_time:
-            self.force_update()
-
-    def force_update(self):
```

Comment:
Did the timer refactoring make it necessary to drop force_update (in python and c++)?

Reply:
My understanding of the original implementation was, that `force_update()` will always update, but `update()` will only call `force_update()` when the period was exceeded. See https://github.com/ros/diagnostics/blob/ros2-devel/diagnostic_updater/diagnostic_updater/_diagnostic_updater.py#L254-L255 

This would mean that the two methods indeed have a different behavior. Calling `update()` with twice the configured rate will _still_ only send updates in the configrued rate, calling `force_update()` with twice the configured rate will _actually_ send updates with twice the configured rate.

Reply:
I think I see what you mean. In https://github.com/ros/diagnostics/pull/114/commits/18f5c771a1ba8cd1b9374583cc0e28f09f6c2c70 I have re-introduced the `force_update` method to make it more explicit. The result of this is that the `update` function becomes private to avoid any confusion.